File: src/main/java/se/gewalli/entities/Order.java
package se.gewalli.entities;


import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;
import java.util.Collection;

public record Order(
        @JsonProperty("id") int id,
        @JsonProperty("customer") Customer customer,
        @JsonProperty("orderDate") Instant orderDate,
        @JsonProperty("products") Collection<Product> products,
        @JsonProperty("version") int version) {
}

File: src/main/java/se/gewalli/entities/Order.kt
package se.gewalli.entities

import java.time.Instant
import java.util.*

class Order(val id: Int, val customer: Customer, val orderDate: Instant, products: Collection<Product>?, version: Int) {
    @JvmField
    val products: Collection<Product>
    val version: Int

    init {
        this.products = Collections.unmodifiableCollection(products)
        this.version = version
    }
}